pkg/tokens/builtin/zone/issuer.go

```diff
@@ -0,0 +1,97 @@
+package zone
+
+import (
+	"context"
+	"time"
+
+	"github.com/thediveo/enumflag"
+
+	"github.com/kumahq/kuma/pkg/core/tokens"
+)
+
+type Token = string
+
+type Identity struct {
+	Zone  string
+	Scope Scope
+}
+
+// TokenIssuer issues Zone Tokens used then for proving identity of the zone egresses.
+// Issued token can be bound by zone name.
+// See pkg/sds/auth/universal/authenticator.go to check algorithm for authentication
+type TokenIssuer interface {
+	Generate(ctx context.Context, identity Identity, validFor time.Duration) (tokens.Token, error)
+}
+
+var _ TokenIssuer = &jwtTokenIssuer{}
+
+func NewTokenIssuer(issuer tokens.Issuer) TokenIssuer {
+	return &jwtTokenIssuer{
+		issuer: issuer,
+	}
+}
+
+type jwtTokenIssuer struct {
+	issuer tokens.Issuer
+}
+
+func (j *jwtTokenIssuer) Generate(ctx context.Context, identity Identity, validFor time.Duration) (Token, error) {
+	claims := &zoneClaims{
+		Zone:  identity.Zone,
+		Scope: identity.Scope,
+	}
+
+	return j.issuer.Generate(ctx, claims, validFor)
+}
+
+type ScopeItem enumflag.Flag
```

Comment:
TBH I'm not a fan of mixing rest/cli stuff with "core" logic. I'd avoid having dependency on `enumflag` here, but it's not a blocker for this PR.

Reply:
addressed